sigstore-java/src/main/java/dev/sigstore/KeylessSigner.java

```diff
@@ -141,20 +201,20 @@ public Builder sigstoreStagingDefaults()
           RekorVerifier.newRekorVerifier(VerificationMaterial.Staging.rekorPublicKey()));
       oidcClient(WebOidcClient.builder().setIssuer(WebOidcClient.STAGING_DEX_ISSUER).build());
       signer(Signers.newEcdsaSigner());
+      minSigningCertificateLifetime(DEFAULT_MIN_SIGNING_CERTIFICATE_LIFETIME);
       return this;
     }
   }
 
   /**
    * Sign one or more artifact digests using the keyless signing workflow. The oidc/fulcio dance to
    * obtain a signing certificate will only occur once. The same ephemeral private key will be used
-   * to sign all artifacts. Errors may occur is the request is for an overwhelming number of
-   * artifactDigests as the certificate may expire -- this method does not current have the ability
-   * to obtain a new certificate if the one is use expires.
+   * to sign all artifacts. This method will renew certificates as they expire.
    *
    * @param artifactDigests sha256 digests of the artifacts to sign.
    * @return a list of keyless singing results.
    */
+  @CheckReturnValue
```

Comment:
So this is now not thread safe? Should we make it threadsafe? I see that you use synchronized somewhere else?

Should we at least add that this is not threadsafe here to the javadoc?

Reply:
Nice catch. I've added `ReentrantReadWriteLock` to make certificate renewal thread-safe.
In the ideal case, we should probably add timeouts for `.lock()` methods and reuse, however, I believe it is fine to go with no timeout for `.lock()`.